coral/src/app/features/topics/list/TopicCard.test.tsx

```diff
@@ -0,0 +1,98 @@
+// @TODO import from "@testing-library/react/pure" disables auto cleanup, remove when main is merged
+import { cleanup, render, screen, within } from "@testing-library/react/pure";
+import { TopicCard } from "src/app/features/topics/list/TopicCard";
+
+const testCard = {
+  topicName: "aivtopic2",
+  description: "Topic description",
+  teamname: "Ospo",
+  environmentsList: ["DEV"],
+};
+
+describe("TopicCard.tsx", () => {
+  describe("renders a card with the required props", () => {
+    const requiredProps = {
+      topicName: testCard.topicName,
+      description: testCard.description,
+      teamname: testCard.teamname,
+      environmentsList: testCard.environmentsList,
+    };
+
+    beforeAll(() => {
+      render(<TopicCard {...requiredProps} />);
+    });
+
+    afterAll(cleanup);
+
+    it("shows a list item ", () => {
+      const listItem = screen.getByRole("listitem");
+
+      expect(listItem).toBeVisible();
+    });
+
+    it("shows the topicName in the list item", () => {
+      const headline = within(screen.getByRole("listitem")).getByRole(
```

Comment:
What values does `within(screen.getByRole("listitem"))...` compared to  `screen.getByRole("heading", { name: requiredProps.topicName })`? I think we could have this with simple `screen.getByRole...`

Reply:
`within(screen.getByRole("listitem"))` makes sure you're querying only for a child element from this listitem. 

Making sure that the heading is part of the listitem is important for accessibility reasons, I decided to test it. 

In most cases you want to avoid making a selector dependent on DOM structure. For me the exception is when the DOM structure is important for having a valid / accessible DOM. 

-> since I moved the `li` out of the component, it's now not relevant for this test anymore, so I'll resolve this comment now. Which is not meant as "discussion is over 🙅‍♀️" of course!

Reply:
> Making sure that the heading is part of the listitem is important for accessibility reasons, I decided to test it.

Fair point. So I approve your approach, but like you said, I think this needs to be changed as we move the semantic value of list item outside of this component.